tests/ui/rust-2024/unsafe-extern-blocks/extern-items.rs

```diff
@@ -0,0 +1,19 @@
+//@ revisions: edition2021 edition2024
+//@[edition2021] edition:2021
+#![cfg_attr(edition2021, feature(unsafe_extern_blocks))]
+//@[edition2024] edition:2024
+//@[edition2024] compile-flags: -Zunstable-options
+//@ check-pass
+
+extern "C" {
+    //~^ WARN extern blocks should be unsafe [missing_unsafe_on_extern]
+    static TEST1: i32;
+    fn test1(i: i32);
+}
```

Comment:
From the edition side, what we would probably expect this PR to do is the following:

- Allow `unsafe extern` in all editions with `#![feature(unsafe_extern_blocks)]`.
- With `#![feature(unsafe_extern_blocks)]`, issue a FCW (at `allow` level) in all editions about `extern` without `unsafe`.
  - Either in this PR or in a separate one, add a machine-applicable fix to this lint.
  - Add this lint to `rust_2024_compatibility`.
- In Rust 2024, with `#![feature(unsafe_extern_blocks)]`, hard error on `extern` without `unsafe`.

The next step would then be a stabilization PR for `unsafe_extern_blocks`.